internal/api_server/server.go

```diff
@@ -0,0 +1,115 @@
+package api_server
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"net/http"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	api "github.com/flightctl/flightctl/api/v1alpha1"
+	"github.com/flightctl/flightctl/internal/api/server"
+	"github.com/flightctl/flightctl/internal/config"
+	"github.com/flightctl/flightctl/internal/crypto"
+	"github.com/flightctl/flightctl/internal/service"
+	"github.com/flightctl/flightctl/internal/store"
+	"github.com/flightctl/flightctl/internal/tasks"
+	"github.com/flightctl/flightctl/pkg/queues"
+	"github.com/go-chi/chi/v5"
+	"github.com/go-chi/chi/v5/middleware"
+	oapimiddleware "github.com/oapi-codegen/nethttp-middleware"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	gracefulShutdownTimeout = 5 * time.Second
+)
+
+type Server struct {
+	log      logrus.FieldLogger
+	cfg      *config.Config
+	store    store.Store
+	ca       *crypto.CA
+	listener net.Listener
+}
+
+// New returns a new instance of a flightctl server.
+func New(
+	log logrus.FieldLogger,
+	cfg *config.Config,
+	store store.Store,
+	ca *crypto.CA,
+	listener net.Listener,
+) *Server {
+	return &Server{
+		log:      log,
+		cfg:      cfg,
+		store:    store,
+		ca:       ca,
+		listener: listener,
+	}
+}
+
+func oapiErrorHandler(w http.ResponseWriter, message string, statusCode int) {
+	http.Error(w, fmt.Sprintf("API Error: %s", message), statusCode)
+}
+
+func (s *Server) Run() error {
+	s.log.Println("Initializing async jobs")
+	provider := queues.NewAmqpProvider(s.cfg.Queue.AmqpURL, s.log)
+	publisher, err := tasks.FleetQueuePublisher(provider)
+	if err != nil {
+		return err
+	}
+	callbackManager := tasks.NewCallbackManager(publisher, s.log)
+
+	s.log.Println("Initializing API server")
+	swagger, err := api.GetSwagger()
+	if err != nil {
+		return fmt.Errorf("failed loading swagger spec: %w", err)
+	}
+	// Skip server name validation
+	swagger.Servers = nil
+
+	oapiOpts := oapimiddleware.Options{
+		ErrorHandler: oapiErrorHandler,
+	}
+
+	router := chi.NewRouter()
+	router.Use(
+		middleware.RequestID,
+		middleware.Logger,
+		middleware.Recoverer,
+		oapimiddleware.OapiRequestValidatorWithOptions(swagger, &oapiOpts),
+	)
+
+	h := service.NewServiceHandler(s.store, callbackManager, s.ca, s.log)
+	server.HandlerFromMux(server.NewStrictHandler(h, nil), router)
```

Comment:
just for awareness... I am splitting the server in two endpoints over here: https://github.com/flightctl/flightctl/pull/303/files

We will need to eventually merge both PRs